### DISCUSSION TITLE
LASERGRBL Laser will not turn on with manual M3 command using button

### DISCUSSION THREAD

**Discussion by ptcruiserx3** [arkypita/LaserGRBL#1071] [Q&A]
i have been using LASERGRBL for 2 years and it works fine for engraving images...however it will not turn laser on via manual command M3

i have tried M3 S255 ..still no laser on

i created a button so i can power on laser to see where my xy location is but it does not work..laser will not turn on

it shows the command in the window when i activate the button but laser does not turn on

laser works fine when doing gcode files in LASERGRBL

not being able to turn the laser on to see starting point makes engraving very difficult.

**Comment by gmmanonymus111** [comment]
Try using `G1F1M3S255` or turn off laser mode.
When lasermode is enabled (`$32=1`) GRBL will only turn the laser on during movement.

**Comment by ptcruiserx3** [comment]
Thanks ill try...however in the lasergrbl faq it does show sample button
for laser on as m3 sxx

I want to use for location purpose only so needs to not burn...its a 2w
laser what s value should i use?

**Comment by ptcruiserx3** [comment]
And what you mean turn off laser mode?...my mschine is similar to
eleksmaker..its chinese knockoff...shouldnt laser mode be activated

**Comment by gmmanonymus111** [comment]
Well, `S2` - `S10` should be enough, depending on the brightness you need
EDIT: To turn off lasermode, send `$32=0`. But you need to set it back to 1 once you positioned your laser. I'll post that custom button along with the " machine move" imitator

**Comment by ptcruiserx3** [comment]
Ok thanks...i try
Its hard to rngrave without knowing start point so spotting is needed..ive
been eyeballing for 2yrs and its frustrsting

The laser on off button done per faq lasergrbl never worked

I hope this works

**Comment by ptcruiserx3** [comment]
And i appreciated the fast response...

**Comment by ptcruiserx3** [comment]
So for my laser button it would look like this

G1 F1 M3 S10

**Comment by gmmanonymus111** [comment]
- Custom button one (disable lasermode):

Type: TwoStateButton
GCode:
```
$32=0
M3 S1
```
GCode2:
```
M5 S0
$32=1
```

- The second one ("fake moving"):

Type: TwoStateButton
GCode:
```
G1 F1 M3 S1 
```
GCode2:
```
G0 M5 S0
```
More buttons can be found [here](https://github.com/arkypita/LaserGRBL/issues/306)

**Comment by ptcruiserx3** [comment]
U r awesome...dat worked

They need change faq button info on lasergrbl site

It burned at s10...so need to change lower ..unless s not working

**Comment by ptcruiserx3** [comment]
It works great...as i change s value my pwm boltage is changing and laser
turning on with button and off so all is good now

Thank you

**Comment by gmmanonymus111** [comment]
Glad I could help

**Comment by ptcruiserx3** [comment]
S2 was giving me about .05v which seems about right

2/255 x 5v = .04v

**Comment by ptcruiserx3** [comment]
See link

https://lasergrbl.com/usage/custom-buttons/

This is what made me scratch my head...

It say m3 Sxx

That no work..i think lasergrbl should revise that info

**Comment by ptcruiserx3** [comment]
LASERGRBL is awesome program thank you

i have one suggestion though

when importing a svg file it does not allow you to resize the output
size  in lasergrbl for diff applications

it would be nice if it could be resized to whatever size needed for
each project...

when you import a jpg it allows output to be resized in lasergrbl
after it has been vectorized but for some reason it wont let you
resize a svg that is imported in

can this be done?

thanks

**Comment by arkypita** [comment]
> They need change faq button info on lasergrbl site

Yes, thanks for the hint.

> when importing a svg file it does not allow you to resize the output
> it would be nice if it could be resized to whatever size needed for

Already in development roadmap

**Comment by arkypita** [comment]
> i have been using LASERGRBL for 2 years and it works fine for engraving images...however it will not turn laser on via manual command M3
> 
> i have tried M3 S255 ..still no laser on

Just for knowledge: this is not an issue of LaserGRBL, it's an issue of GRBL!
Their [documentation about M3 in Laser Mode](https://github.com/gnea/grbl/wiki/Grbl-v1.1-Laser-Mode) say:

![image](https://user-images.githubusercontent.com/8782035/87348416-f8c7f480-c554-11ea-9c93-b36a08813a09.png)

So, in my opinion, M3 S10 should be enough to turn on laser in low power.

**Comment by Ggmax4d** [comment]
HI folks
One thing I have is a 200 mHz scope DSO38 by Jyetech, enough 4 me. Then I put it between ground and Z pin (+ or -), that shows me if a square signal is sent to the laser. That helped me to detect some troubleshootings that, in fact came from most of the time bad electrical contacts.

**Comment by ptcruiserx3** [comment]
I found issue...it was incorrect ttl connection...it fixed now

**Comment by ptcruiserx3** [comment]
I ust ordered a cheap ebay k40 laser...i have heard for the money they
are ok...i think i have to use k40 whisperer on that one.

**Comment by crazy999666676** [comment]
I could fix the focus problem but now I have no more light when the laser drive aroubd the frame.  I used the option to align the workpiece. Thx for help
